Unable to fork /bin/clang.

**Describe the bug**
While I am setting up retina in K8s infra. I am facing below error. 

```ts=2024-03-23T16:55:10.369Z level=debug caller=loader/compile.go:26 msg="Error running command" goversion=go1.21.8 os=linux arch=arm64 numcores=8 hostname=ip-10-149-82-88.ec2.internal podname=retina-agent-4hldd version=v0.0.1 apiserver=https://172.20.0.1:443 plugins=packetforward command="/bin/clang -target bpf -Wall -D__TARGET_ARCH_arm64 -g -O2 -c /go/src/github.com/microsoft/retina/pkg/plugin/packetforward/_cprog/packetforward.c -o /go/src/github.com/microsoft/retina/pkg/plugin/packetforward/packetforward_bpf.o -I/go/src/github.com/microsoft/retina/pkg/plugin/packetforward/../lib/_arm64 -I/go/src/github.com/microsoft/retina/pkg/plugin/packetforward/../lib/common/libbpf/_src" stderr= error="fork/exec /bin/clang: no such file or directory"
ts=2024-03-23T16:55:10.369Z level=info caller=server/server.go:79 msg="gracefully shutting down HTTP server..." goversion=go1.21.8 os=linux arch=arm64 numcores=8 hostname=ip-10-149-82-88.ec2.internal podname=retina-agent-4hldd version=v0.0.1 apiserver=https://172.20.0.1:443 plugins=packetforward
ts=2024-03-23T16:55:10.369Z level=info caller=server/server.go:71 msg="HTTP server stopped with err: http: Server closed" goversion=go1.21.8 os=linux arch=arm64 numcores=8 hostname=ip-10-149-82-88.ec2.internal podname=retina-agent-4hldd version=v0.0.1 apiserver=https://172.20.0.1:443 plugins=packetforward
ts=2024-03-23T16:55:10.369Z level=panic caller=controllermanager/controllermanager.go:118 msg="Error running controller manager" goversion=go1.21.8 os=linux arch=arm64 numcores=8 hostname=ip-10-149-82-88.ec2.internal podname=retina-agent-4hldd version=v0.0.1 apiserver=https://172.20.0.1:443 plugins=packetforward error="failed to reconcile plugin packetforward: fork/exec /bin/clang: no such file or directory" errorVerbose="fork/exec /bin/clang: no such file or directory\nfailed to reconcile plugin packetforward\ngithub.com/microsoft/retina/pkg/managers/pluginmanager.(*PluginManager).Start\n\t/go/src/github.com/microsoft/retina/pkg/managers/pluginmanager/pluginmanager.go:169\ngithub.com/microsoft/retina/pkg/managers/controllermanager.(*Controller).Start.func1\n\t/go/src/github.com/microsoft/retina/pkg/managers/controllermanager/controllermanager.go:108\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\t/go/pkg/mod/golang.org/x/sync@v0.6.0/errgroup/errgroup.go:78\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_arm64.s:1197"
panic: Error running controller manager [recovered]
	panic: Error running controller manager

goroutine 102 [running]:
github.com/microsoft/retina/pkg/telemetry.TrackPanic()
	/go/src/github.com/microsoft/retina/pkg/telemetry/telemetry.go:112 +0x1e8
panic({0x1eb6b00?, 0x4000334180?})
	/usr/local/go/src/runtime/panic.go:914 +0x218
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x0?, 0x1?, {0x4000aaace8?, 0x0?, 0x0?})
	/go/pkg/mod/go.uber.org/zap@v1.26.0/zapcore/entry.go:196 +0x78
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0x40002341a0, {0x400007ff00, 0x1, 0x1})
	/go/pkg/mod/go.uber.org/zap@v1.26.0/zapcore/entry.go:262 +0x2c0
go.uber.org/zap.(*Logger).Panic(0x40005f2080?, {0x25cf876?, 0x0?}, {0x400007ff00, 0x1, 0x1})
	/go/pkg/mod/go.uber.org/zap@v1.26.0/logger.go:284 +0x54
github.com/microsoft/retina/pkg/managers/controllermanager.(*Controller).Start(0x4000cbf450, {0x298b488?, 0x400046f450?})
	/go/src/github.com/microsoft/retina/pkg/managers/controllermanager/controllermanager.go:118 +0x22c
created by main.main in goroutine 1
	/go/src/github.com/microsoft/retina/controller/main.go:286 +0x2190
```

- OS: AmzonLinux
- Kubernetes Version: 1.28
- Host: EKS
- Retina Version: 1.0.0

I am not running any operator. for now I am running only DaemonSet and in configmap I am adding plugins 

```
config.yaml: |-
    apiServer:
      host: 0.0.0.0
      port: 10093
    logLevel: debug
    enabledPlugin: ["packetforward"]
    metricsInterval: 10
    enableTelemetry: false
    enablePodLevel: false
    remoteContext: false
    enableAnnotations: false
```

The same thing happening to me as well. 

`ts=2024-03-23T17:54:50.933Z level=info caller=linuxutil/linuxutil_linux.go:42 msg="Initializing linuxutil plugin..." goversion=go1.21.8 os=linux arch=amd64 numcores=2 hostname=kube02 podname=retina-agent-bsbx6 version=v0.0.1 apiserver=https://10.96.0.1:443 plugins=dropreason,packetforward,linuxutil,dns
ts=2024-03-23T17:54:50.933Z level=info caller=pluginmanager/pluginmanager.go:122 msg="Reconciled plugin" goversion=go1.21.8 os=linux arch=amd64 numcores=2 hostname=kube02 podname=retina-agent-bsbx6 version=v0.0.1 apiserver=https://10.96.0.1:443 plugins=dropreason,packetforward,linuxutil,dns name=linuxutil
ts=2024-03-23T17:54:50.933Z level=info caller=pluginmanager/pluginmanager.go:173 msg="starting plugin linuxutil" goversion=go1.21.8 os=linux arch=amd64 numcores=2 hostname=kube02 podname=retina-agent-bsbx6 version=v0.0.1 apiserver=https://10.96.0.1:443 plugins=dropreason,packetforward,linuxutil,dns
ts=2024-03-23T17:54:50.933Z level=info caller=linuxutil/linuxutil_linux.go:57 msg="Running linuxutil plugin..." goversion=go1.21.8 os=linux arch=amd64 numcores=2 hostname=kube02 podname=retina-agent-bsbx6 version=v0.0.1 apiserver=https://10.96.0.1:443 plugins=dropreason,packetforward,linuxutil,dns
ts=2024-03-23T17:54:50.933Z level=info caller=server/server.go:69 msg="starting HTTP server... on " goversion=go1.21.8 os=linux arch=amd64 numcores=2 hostname=kube02 podname=retina-agent-bsbx6 version=v0.0.1 apiserver=https://10.96.0.1:443 plugins=dropreason,packetforward,linuxutil,dns addr=0.0.0.0:10093
ts=2024-03-23T17:54:53.434Z level=info caller=dns/dns_linux.go:89 msg="Stopped dns plugin" goversion=go1.21.8 os=linux arch=amd64 numcores=2 hostname=kube02 podname=retina-agent-bsbx6 version=v0.0.1 apiserver=https://10.96.0.1:443 plugins=dropreason,packetforward,linuxutil,dns
ts=2024-03-23T17:54:53.674Z level=info caller=dns/dns_linux.go:64 msg="Initialized dns plugin" goversion=go1.21.8 os=linux arch=amd64 numcores=2 hostname=kube02 podname=retina-agent-bsbx6 version=v0.0.1 apiserver=https://10.96.0.1:443 plugins=dropreason,packetforward,linuxutil,dns
ts=2024-03-23T17:54:53.674Z level=info caller=pluginmanager/pluginmanager.go:122 msg="Reconciled plugin" goversion=go1.21.8 os=linux arch=amd64 numcores=2 hostname=kube02 podname=retina-agent-bsbx6 version=v0.0.1 apiserver=https://10.96.0.1:443 plugins=dropreason,packetforward,linuxutil,dns name=dns
ts=2024-03-23T17:54:53.674Z level=info caller=pluginmanager/pluginmanager.go:173 msg="starting plugin dns" goversion=go1.21.8 os=linux arch=amd64 numcores=2 hostname=kube02 podname=retina-agent-bsbx6 version=v0.0.1 apiserver=https://10.96.0.1:443 plugins=dropreason,packetforward,linuxutil,dns
ts=2024-03-23T17:54:53.674Z level=error caller=dns/dns_linux.go:76 msg="Failed to attach tracer" goversion=go1.21.8 os=linux arch=amd64 numcores=2 hostname=kube02 podname=retina-agent-bsbx6 version=v0.0.1 apiserver=https://10.96.0.1:443 plugins=dropreason,packetforward,linuxutil,dns error="creating network tracer attachment for pid 1: opening raw socket: operation not permitted"
ts=2024-03-23T17:54:53.674Z level=info caller=linuxutil/linuxutil_linux.go:64 msg="Context is done, linuxutil will stop running" goversion=go1.21.8 os=linux arch=amd64 numcores=2 hostname=kube02 podname=retina-agent-bsbx6 version=v0.0.1 apiserver=https://10.96.0.1:443 plugins=dropreason,packetforward,linuxutil,dns
ts=2024-03-23T17:54:56.175Z level=info caller=server/server.go:79 msg="gracefully shutting down HTTP server..." goversion=go1.21.8 os=linux arch=amd64 numcores=2 hostname=kube02 podname=retina-agent-bsbx6 version=v0.0.1 apiserver=https://10.96.0.1:443 plugins=dropreason,packetforward,linuxutil,dns
ts=2024-03-23T17:54:56.175Z level=info caller=server/server.go:71 msg="HTTP server stopped with err: http: Server closed" goversion=go1.21.8 os=linux arch=amd64 numcores=2 hostname=kube02 podname=retina-agent-bsbx6 version=v0.0.1 apiserver=https://10.96.0.1:443 plugins=dropreason,packetforward,linuxutil,dns
ts=2024-03-23T17:54:56.175Z level=panic caller=controllermanager/controllermanager.go:118 msg="Error running controller manager" goversion=go1.21.8 os=linux arch=amd64 numcores=2 hostname=kube02 podname=retina-agent-bsbx6 version=v0.0.1 apiserver=https://10.96.0.1:443 plugins=dropreason,packetforward,linuxutil,dns error="failed to reconcile plugin dropreason: context canceled" errorVerbose="context canceled\nfailed to reconcile plugin dropreason\ngithub.com/microsoft/retina/pkg/managers/pluginmanager.(*PluginManager).Start\n\t/go/src/github.com/microsoft/retina/pkg/managers/pluginmanager/pluginmanager.go:169\ngithub.com/microsoft/retina/pkg/managers/controllermanager.(*Controller).Start.func1\n\t/go/src/github.com/microsoft/retina/pkg/managers/controllermanager/controllermanager.go:108\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\t/go/pkg/mod/golang.org/x/sync@v0.6.0/errgroup/errgroup.go:78\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1650"
panic: Error running controller manager [recovered]
        panic: Error running controller manager

goroutine 16 [running]:
github.com/microsoft/retina/pkg/telemetry.TrackPanic()
        /go/src/github.com/microsoft/retina/pkg/telemetry/telemetry.go:112 +0x209
panic({0x242fc60?, 0xc0004241a0?})
        /usr/local/go/src/runtime/panic.go:914 +0x21f
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x1?, 0x0?, {0x0?, 0x0?, 0xc000a6aee0?})
        /go/pkg/mod/go.uber.org/zap@v1.26.0/zapcore/entry.go:196 +0x54
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0008b6dd0, {0xc00055f740, 0x1, 0x1})
        /go/pkg/mod/go.uber.org/zap@v1.26.0/zapcore/entry.go:262 +0x3ec
go.uber.org/zap.(*Logger).Panic(0xc00055e2c0?, {0x2b48afa?, 0x0?}, {0xc00055f740, 0x1, 0x1})
        /go/pkg/mod/go.uber.org/zap@v1.26.0/logger.go:284 +0x51
github.com/microsoft/retina/pkg/managers/controllermanager.(*Controller).Start(0xc000a7abe0, {0x2f057d0?, 0xc0006afb30?})
        /go/src/github.com/microsoft/retina/pkg/managers/controllermanager/controllermanager.go:118 +0x28c
created by main.main in goroutine 1
        /go/src/github.com/microsoft/retina/controller/main.go:286 +0x2825
~/Projects/retina (main ✔)`

Host: ubuntu 22.04.3
Kubernetes Version:  v1.28.6

hello, i encountered the same problem after recompiling retina controller image locally.
Compilation environment：darwin ARM64
Running environment: 
- OS: CentOS Linux 7 (Core)
- Kernel Version: 5.4.207-1.el7.elrepo.x86_64
- Kubernetes Version: 1.22.2
- Retina Version: v0.0.5

retina agent error logs:
```
ts=2024-04-09T05:55:40.697Z level=debug caller=loader/compile.go:22 msg=Running goversion=go1.21.9 os=linux arch=amd64 numcores=48 hostname=node4 podname=retina-agent-ppjlb version=v0.0.5 apiserver=https://10.68.0.1:443 plugins=dropreason,packetforward,linuxutil,dns command="/bin/clang -target bpf -Wall -D__TARGET_ARCH_x86 -g -O2 -c /go/src/github.com/microsoft/retina/pkg/plugin/dropreason/_cprog/drop_reason.c -o /go/src/github.com/microsoft/retina/pkg/plugin/dropreason/kprobe_bpf.o -I/go/src/github.com/microsoft/retina/pkg/plugin/dropreason/../lib/_amd64 -I/go/src/github.com/microsoft/retina/pkg/plugin/dropreason/../lib/common/libbpf/_src -I/go/src/github.com/microsoft/retina/pkg/plugin/dropreason/../filter/_cprog/"
ts=2024-04-09T05:55:40.697Z level=debug caller=loader/compile.go:26 msg="Error running command" goversion=go1.21.9 os=linux arch=amd64 numcores=48 hostname=node4 podname=retina-agent-ppjlb version=v0.0.5 apiserver=https://10.68.0.1:443 plugins=dropreason,packetforward,linuxutil,dns command="/bin/clang -target bpf -Wall -D__TARGET_ARCH_x86 -g -O2 -c /go/src/github.com/microsoft/retina/pkg/plugin/dropreason/_cprog/drop_reason.c -o /go/src/github.com/microsoft/retina/pkg/plugin/dropreason/kprobe_bpf.o -I/go/src/github.com/microsoft/retina/pkg/plugin/dropreason/../lib/_amd64 -I/go/src/github.com/microsoft/retina/pkg/plugin/dropreason/../lib/common/libbpf/_src -I/go/src/github.com/microsoft/retina/pkg/plugin/dropreason/../filter/_cprog/" stderr= error="fork/exec /bin/clang: no such file or directory"
ts=2024-04-09T05:55:40.697Z level=info caller=linuxutil/linuxutil_linux.go:64 msg="Context is done, linuxutil will stop running" goversion=go1.21.9 os=linux arch=amd64 numcores=48 hostname=node4 podname=retina-agent-ppjlb version=v0.0.5 apiserver=https://10.68.0.1:443 plugins=dropreason,packetforward,linuxutil,dns
ts=2024-04-09T05:55:40.697Z level=info caller=server/server.go:79 msg="gracefully shutting down HTTP server..." goversion=go1.21.9 os=linux arch=amd64 numcores=48 hostname=node4 podname=retina-agent-ppjlb version=v0.0.5 apiserver=https://10.68.0.1:443 plugins=dropreason,packetforward,linuxutil,dns
ts=2024-04-09T05:55:40.697Z level=info caller=watchermanager/watchermanager.go:71 msg="watcher stopping..." goversion=go1.21.9 os=linux arch=amd64 numcores=48 hostname=node4 podname=retina-agent-ppjlb version=v0.0.5 apiserver=https://10.68.0.1:443 plugins=dropreason,packetforward,linuxutil,dns watcher_type=*apiserver.ApiServerWatcher
ts=2024-04-09T05:55:40.697Z level=info caller=watchermanager/watchermanager.go:71 msg="watcher stopping..." goversion=go1.21.9 os=linux arch=amd64 numcores=48 hostname=node4 podname=retina-agent-ppjlb version=v0.0.5 apiserver=https://10.68.0.1:443 plugins=dropreason,packetforward,linuxutil,dns watcher_type=*endpoint.EndpointWatcher
ts=2024-04-09T05:55:40.697Z level=info caller=server/server.go:71 msg="HTTP server stopped with err: http: Server closed" goversion=go1.21.9 os=linux arch=amd64 numcores=48 hostname=node4 podname=retina-agent-ppjlb version=v0.0.5 apiserver=https://10.68.0.1:443 plugins=dropreason,packetforward,linuxutil,dns
ts=2024-04-09T05:55:40.697Z level=panic caller=controllermanager/controllermanager.go:119 msg="Error running controller manager" goversion=go1.21.9 os=linux arch=amd64 numcores=48 hostname=node4 podname=retina-agent-ppjlb version=v0.0.5 apiserver=https://10.68.0.1:443 plugins=dropreason,packetforward,linuxutil,dns error="failed to reconcile plugin dropreason: fork/exec /bin/clang: no such file or directory" errorVerbose="fork/exec /bin/clang: no such file or directory\nfailed to reconcile plugin dropreason\ngithub.com/microsoft/retina/pkg/managers/pluginmanager.(*PluginManager).Start\n\t/go/src/github.com/microsoft/retina/pkg/managers/pluginmanager/pluginmanager.go:169\ngithub.com/microsoft/retina/pkg/managers/controllermanager.(*Controller).Start.func1\n\t/go/src/github.com/microsoft/retina/pkg/managers/controllermanager/controllermanager.go:109\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\t/go/pkg/mod/golang.org/x/sync@v0.6.0/errgroup/errgroup.go:78\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1650"
panic: Error running controller manager

goroutine 415 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x1?, 0x0?, {0x0?, 0x0?, 0xc002294020?})
	/go/pkg/mod/go.uber.org/zap@v1.26.0/zapcore/entry.go:196 +0x54
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc000a75860, {0xc001e817c0, 0x1, 0x1})
	/go/pkg/mod/go.uber.org/zap@v1.26.0/zapcore/entry.go:262 +0x3ec
go.uber.org/zap.(*Logger).Panic(0xc000eaf600?, {0x2b62b8b?, 0x0?}, {0xc001e817c0, 0x1, 0x1})
	/go/pkg/mod/go.uber.org/zap@v1.26.0/logger.go:284 +0x51
github.com/microsoft/retina/pkg/managers/controllermanager.(*Controller).Start(0xc0001c1180, {0x2f22170?, 0xc0005a8000?})
	/go/src/github.com/microsoft/retina/pkg/managers/controllermanager/controllermanager.go:119 +0x28c
created by main.main in goroutine 1
	/go/src/github.com/microsoft/retina/controller/main.go:290 +0x28d0
```

and i have confirmed that the /bin/clang file exists in the retina controller image.

--- a/controller/Dockerfile.controller
+++ b/controller/Dockerfile.controller
@@ -2,7 +2,7 @@ ARG builderImage="ghcr.io/microsoft/retina/retina-builder:0.0.1"
 ARG toolsImage="ghcr.io/microsoft/retina/retina-tools:0.0.1"
 
 # Stage: Build binary
-FROM --platform=$TARGETPLATFORM mcr.microsoft.com/oss/go/microsoft/golang:1.21 AS builder
+FROM --platform=$BUILDPLATFORM mcr.microsoft.com/oss/go/microsoft/golang:1.21 AS builder
 LABEL Name=retina-builder Version=0.0.1
 
 RUN apt-get update &&\
@@ -44,16 +44,14 @@ RUN --mount=type=cache,target="/root/.cache/go-build" go build -v -o /go/bin/ret
 
 # Stage: Prepare clang and tools
 # Bullseye -> debian11
-FROM  --platform=$TARGETPLATFORM mcr.microsoft.com/mirror/docker/library/debian:bullseye@sha256:a648e10e02af129706b1fb89e1ac9694ae3db7f2b8439aa906321e68cc281bc0 AS tools
+FROM  --platform=$BUILDPLATFORM mcr.microsoft.com/mirror/docker/library/debian:bullseye@sha256:a648e10e02af129706b1fb89e1ac9694ae3db7f2b8439aa906321e68cc281bc0 AS tools
 
 LABEL Name=retina-tools Version=0.0.1
 
 WORKDIR /tmp
 RUN apt-get update && \
     apt-get install -y --no-install-recommends \
-    curl xz-utils binutils wget gnupg2
-RUN apt-get install -y --no-install-recommends \
-    ca-certificates tcpdump iproute2 iptables
+    curl xz-utils binutils wget gnupg2 ca-certificates
 
 RUN mkdir -p /usr/local
 
@@ -69,16 +67,21 @@ RUN if [ "$GOARCH" = "amd64" ] ; then \
     gpg2 --verify clang+llvm.tar.xz.sig clang+llvm.tar.xz; \
     tar -C /usr/local -xJf ./clang+llvm.tar.xz --no-same-owner; \
     mv /usr/local/clang+llvm-14.0.0-x86_64-linux-gnu-ubuntu-18.04 /usr/local/clang+llvm; \
+    dpkg --add-architecture amd64; \
+    apt-get update; \
+    apt-get install -y --no-install-recommends tcpdump:amd64 iproute2:amd64 iptables:amd64; \
     else \
     # GOARCH=Arm64.
     # Download clang and llvm.
     # Need more dependencies for Arm64.
+    dpkg --add-architecture arm64; \
     apt-get update; \
-    apt-get install -y --no-install-recommends libc6 zlib1g libncurses5; \
+    apt-get install -y --no-install-recommends libc6:arm64 zlib1g:arm64 libncurses5:arm64; \
     apt-get install -y --no-install-recommends apt-file; \
     apt-file update; \
     apt-file find libstdc++.so.6; \
-    apt-get install -y --no-install-recommends libstdc++6; \
+    apt-get install -y --no-install-recommends libstdc++6:arm64; \
+    apt-get install -y --no-install-recommends tcpdump:arm64 iproute2:arm64 iptables:arm64; \
     # Download clang and llvm.
     wget -O clang+llvm.tar.xz https://github.com/llvm/llvm-project/releases/download/llvmorg-14.0.0/clang+llvm-14.0.0-aarch64-linux-gnu.tar.xz; \
     tar -C /usr/local -xJf ./clang+llvm.tar.xz --no-same-owner; \

--- a/controller/Dockerfile.controller
+++ b/controller/Dockerfile.controller
@@ -2,7 +2,7 @@ ARG builderImage="ghcr.io/microsoft/retina/retina-builder:0.0.1"
 ARG toolsImage="ghcr.io/microsoft/retina/retina-tools:0.0.1"
 
 # Stage: Build binary
-FROM --platform=$BUILDPLATFORM mcr.microsoft.com/oss/go/microsoft/golang:1.21 AS builder
+FROM --platform=$TARGETPLATFORM mcr.microsoft.com/oss/go/microsoft/golang:1.21 AS builder
 LABEL Name=retina-builder Version=0.0.1
 
 RUN apt-get update &&\
@@ -44,14 +44,16 @@ RUN --mount=type=cache,target="/root/.cache/go-build" go build -v -o /go/bin/ret
 
 # Stage: Prepare clang and tools
 # Bullseye -> debian11
-FROM  --platform=$BUILDPLATFORM mcr.microsoft.com/mirror/docker/library/debian:bullseye@sha256:a648e10e02af129706b1fb89e1ac9694ae3db7f2b8439aa906321e68cc281bc0 AS tools
+FROM  --platform=$TARGETPLATFORM mcr.microsoft.com/mirror/docker/library/debian:bullseye@sha256:a648e10e02af129706b1fb89e1ac9694ae3db7f2b8439aa906321e68cc281bc0 AS tools
 
 LABEL Name=retina-tools Version=0.0.1
 
 WORKDIR /tmp
 RUN apt-get update && \
     apt-get install -y --no-install-recommends \
-    curl xz-utils binutils wget gnupg2 ca-certificates
+    curl xz-utils binutils wget gnupg2
+RUN apt-get install -y --no-install-recommends \
+    ca-certificates tcpdump iproute2 iptables
 
 RUN mkdir -p /usr/local
 
@@ -67,21 +69,15 @@ RUN if [ "$GOARCH" = "amd64" ] ; then \
     gpg2 --verify clang+llvm.tar.xz.sig clang+llvm.tar.xz; \
     tar -C /usr/local -xJf ./clang+llvm.tar.xz --no-same-owner; \
     mv /usr/local/clang+llvm-14.0.0-x86_64-linux-gnu-ubuntu-18.04 /usr/local/clang+llvm; \
-    dpkg --add-architecture amd64; \
-    apt-get update; \
-    apt-get install -y --no-install-recommends tcpdump:amd64 iproute2:amd64 iptables:amd64; \
     else \
     # GOARCH=Arm64.
     # Download clang and llvm.
     # Need more dependencies for Arm64.
-    dpkg --add-architecture arm64; \
-    apt-get update; \
-    apt-get install -y --no-install-recommends libc6:arm64 zlib1g:arm64 libncurses5:arm64; \
+    apt-get install -y --no-install-recommends libc6 zlib1g libncurses5; \
     apt-get install -y --no-install-recommends apt-file; \
     apt-file update; \
     apt-file find libstdc++.so.6; \
-    apt-get install -y --no-install-recommends libstdc++6:arm64; \
-    apt-get install -y --no-install-recommends tcpdump:arm64 iproute2:arm64 iptables:arm64; \
+    apt-get install -y --no-install-recommends libstdc++6; \
     # Download clang and llvm.
     wget -O clang+llvm.tar.xz https://github.com/llvm/llvm-project/releases/download/llvmorg-14.0.0/clang+llvm-14.0.0-aarch64-linux-gnu.tar.xz; \
     tar -C /usr/local -xJf ./clang+llvm.tar.xz --no-same-owner; \

--- a/controller/Dockerfile.controller
+++ b/controller/Dockerfile.controller
@@ -2,7 +2,7 @@ ARG builderImage="ghcr.io/microsoft/retina/retina-builder:0.0.1"
 ARG toolsImage="ghcr.io/microsoft/retina/retina-tools:0.0.1"
 
 # Stage: Build binary
-FROM --platform=$BUILDPLATFORM mcr.microsoft.com/oss/go/microsoft/golang:1.21 AS builder
+FROM --platform=$TARGETPLATFORM mcr.microsoft.com/oss/go/microsoft/golang:1.21 AS builder
 LABEL Name=retina-builder Version=0.0.1
 
 RUN apt-get update &&\
@@ -44,7 +44,7 @@ RUN --mount=type=cache,target="/root/.cache/go-build" go build -v -o /go/bin/ret
 
 # Stage: Prepare clang and tools
 # Bullseye -> debian11
-FROM  --platform=$BUILDPLATFORM mcr.microsoft.com/mirror/docker/library/debian:bullseye@sha256:a648e10e02af129706b1fb89e1ac9694ae3db7f2b8439aa906321e68cc281bc0 AS tools
+FROM  --platform=$TARGETPLATFORM mcr.microsoft.com/mirror/docker/library/debian:bullseye@sha256:a648e10e02af129706b1fb89e1ac9694ae3db7f2b8439aa906321e68cc281bc0 AS tools
 
 LABEL Name=retina-tools Version=0.0.1
 
@@ -73,13 +73,12 @@ RUN if [ "$GOARCH" = "amd64" ] ; then \
     # GOARCH=Arm64.
     # Download clang and llvm.
     # Need more dependencies for Arm64.
-    dpkg --add-architecture arm64; \
     apt-get update; \
-    apt-get install -y --no-install-recommends libc6:arm64 zlib1g:arm64 libncurses5:arm64; \
+    apt-get install -y --no-install-recommends libc6 zlib1g libncurses5; \
     apt-get install -y --no-install-recommends apt-file; \
     apt-file update; \
     apt-file find libstdc++.so.6; \
-    apt-get install -y --no-install-recommends libstdc++6:arm64; \
+    apt-get install -y --no-install-recommends libstdc++6; \
     # Download clang and llvm.
     wget -O clang+llvm.tar.xz https://github.com/llvm/llvm-project/releases/download/llvmorg-14.0.0/clang+llvm-14.0.0-aarch64-linux-gnu.tar.xz; \
     tar -C /usr/local -xJf ./clang+llvm.tar.xz --no-same-owner; \
@@ -98,7 +97,7 @@ RUN arr="tcpdump ip ss iptables-legacy iptables-legacy-save iptables-nft iptable
 # ----------------------------------------------------------------------------------- #
 
 # Stage: Base distroless init image
-FROM --platform=$BUILDPLATFORM mcr.microsoft.com/mirror/gcr/distroless/cc-debian11:latest@sha256:b53fbf5f81f4a120a489fedff2092e6fcbeacf7863fce3e45d99cc58dc230ccc as init
+FROM --platform=$TARGETPLATFORM mcr.microsoft.com/mirror/gcr/distroless/cc-debian11:latest@sha256:b53fbf5f81f4a120a489fedff2092e6fcbeacf7863fce3e45d99cc58dc230ccc as init
 LABEL Name=retina-init Version=0.0.1
 
 COPY --from=builder /go/bin/retina/initretina /retina/initretina
@@ -113,7 +112,7 @@ ENTRYPOINT ["./retina/initretina"]
 # ----------------------------------------------------------------------------------- #
 
 # Stage: Base distroless image
-FROM --platform=$BUILDPLATFORM mcr.microsoft.com/mirror/gcr/distroless/cc-debian11:latest@sha256:b53fbf5f81f4a120a489fedff2092e6fcbeacf7863fce3e45d99cc58dc230ccc as agent
+FROM --platform=$TARGETPLATFORM mcr.microsoft.com/mirror/gcr/distroless/cc-debian11:latest@sha256:b53fbf5f81f4a120a489fedff2092e6fcbeacf7863fce3e45d99cc58dc230ccc as agent
 
 LABEL Name=retina-controller Version=0.0.1
 

